Docs: move in section "Topics - Processes" (#4048)
aiidateam/aiida-core: docs/source/topics/processes/index.rst

.. _topics:processes:

*********
Processes
*********

This topic section provides extensive and detailed information on the concept of processes in AiiDA and how to work with them.

> Since the concepts explained here apply to processes in general, they also apply to its subtypes, calculations and workflows.

Reviewer: ```suggestion
Since the concepts explained here apply to processes in general, they also apply to its subtypes: calculations and workflows.
```

.. _topics:engine:processes:

*********
Processes
*********

> Before you start working with processes, make sure you have read and understood the :ref:`basic concept<topics:processes:concepts>`.

Reviewer: I don't think this is in the spirit of a background section that we imply that users need to have read to be able to work with processes.

I am also not a fan of this slightly aggressive language that is telling users what to do. I think it's better to use slightly more friendly language that is providing hints and suggestions instead or simply states facts like this:
```suggestion
This section assumes knowledge of the :ref:`basic concepts<topics:processes:concepts>` .
```
We could also add this as a `note::`.

This section will explain the aspects of working with processes that apply to all the various types of processes.
Details that only pertain to a specific sub type of process, will be documented in their respective sections:

    * :ref:`calculation functions<topics:calculations:usage:calcfunctions>`
    * :ref:`calculation jobs<topics:calculations:usage:calcjobs>`
    * :ref:`work functions<topics:workflows:usage:workfunctions>`
    * :ref:`work chains<topics:workflows:usage:workchains>`

> Since all of these are types of processes, everything that will be explained in this section, will apply to each and everyone of them.
> That makes it very useful to read and understand this section well, as the concepts apply so broadly.
> However, for the same reason, at times this section may feel a bit abstract.
> It may therefore be advisable to start reading a section on one of the more specific processes listed above first, to get a more concrete example, and then simply refer back here for a more extensive explanation of the details.

Reviewer: I appreciate providing guidance to the reader, but the language feels a bit imposing. I'd prefer to stick to more factual statements instead and then let the reader draw their own conclusions:
```suggestion
All of the types listed above are processes, meaning they are concrete implementations of the more abstract concepts described in this section.
```
Again, might be worthwhile to format all of the "guidance for readers" notes as `notes::`. We could even create a distinct sphinx directive for that, but I'm trying to keep things as simple as possible during the transition, so probably better to just stick to `notes::` for now.
